src/loom-provider.ts

```diff
@@ -280,8 +280,9 @@ export class LoomProvider {
     return this.accountsAddrList
   }
 
-  private _ethBlockNumber() {
-    return numberToHex(+this._client.getBlockHeightAsync())
+  private async _ethBlockNumber() {
+    const blockNumber = await this._client.getBlockHeightAsync()
+    return numberToHex(+blockNumber)
```

Comment:
`getBlockHeightAsync()` returns `Promise<number>` there should be no implicit conversion necessary here (`+blockNumber`).